src/compat/db_json.cpp

```diff
@@ -708,6 +743,57 @@ db_json_add_element_to_array (JSON_DOC *doc, const JSON_DOC *value)
   doc->PushBack (new_doc, doc->GetAllocator ());
 }
 
+static bool
+db_json_contains_duplicate_keys (const JSON_DOC &doc, std::string &duplicate_key)
+{
+  duplicate_key = "";
+  JSON_POINTER p ("");
+  const JSON_VALUE *head = p.Get (doc);
+  // we will iterate through json document and check if the object has duplicate keys
+  std::queue<const JSON_VALUE *> mqueue;
+  std::unordered_set<std::string> inserted_keys;
+
+  // add the head first
+  mqueue.push (head);
+
+  while (!mqueue.empty())
+    {
+      // get the current element
+      const JSON_VALUE *current = mqueue.front();
+      mqueue.pop();
+
+      // iterate through the array or object and push elements into queue
+      if (current->IsArray())
+	{
+	  for (auto it = current->GetArray().begin(); it != current->GetArray().end(); ++it)
+	    {
+	      mqueue.push (it);
+	    }
+	}
+      else if (current->IsObject())
+	{
+	  // we need to clear the set because we check only keys from the same object (on the same level)
+	  inserted_keys.clear();
+	  // iterate through object`s keys and verify if it has duplicate keys
+	  for (auto it = current->MemberBegin(); it != current->MemberEnd(); ++it)
+	    {
+	      const std::string &key = it->name.GetString();
+	      if (inserted_keys.find (key) != inserted_keys.end())
+		{
+		  // we found a duplicate key
+		  duplicate_key = key;
+		  return true;
+		}
+
+	      inserted_keys.insert (key);
```

Comment:
I believe insert makes an unnecessary copy of key. how about calling with std::move (key) ?